MitoMathHelper.js

```diff
@@ -172,4 +172,94 @@ const MitoMathHelper = class MitoMathHelper {
         let normalizedDistance = aToPDoToAToB / aToBSquared;
         return [linePointA[0] + (aToB[0] * normalizedDistance), linePointA[1] + (aToB[1] * normalizedDistance)]
     }
+
+    /**
+     * returns magnitude of a motion
+     * @param vector
+     * @returns {number}
+     * @private
+     */
+    static _getMagnitude(vector) {
+        return Math.hypot(vector[0], vector[1]);
+    }
+
+    /**
+     * returns magnitude squared
+     * @param vector
+     * @returns {number}
+     * @private
+     */
+    static _getMagnitudeSquared(vector) {
+        return vector[0] * vector[0] + vector[1] * vector[1];
+    }
+
+    /**
+     * returns dot product
+     * @param vectorA
+     * @param vectorB
+     * @returns {number}
+     * @private
+     */
+    static _dotProduct(vectorA, vectorB) {
+        return vectorA[0] * vectorB[0] + vectorA[1] * vectorB[1];
+    }
+
+    /**
+     * returns angle in radians between two bodies
+     * @param bodyA
+     * @param bodyB
+     * @returns {number}
+     * @private
+     */
+    static _getAngleBetweenBodies(bodyA, bodyB) {
+        let positionA = bodyA.getPosition();
+        let positionB = bodyB.getPosition();
+        return Math.atan2(positionB[1] - positionA[1], positionB[0] - positionA[0]);
+    }
+
+    /** uses that math formula to find translational velocity of one vector
+     * @param massA
+     * @param posA
+     * @param velocityA
+     * @param massB
+     * @param posB
+     * @param velocityB
+     * @returns {[]} resulting velocity
+     * @private
+     */
+    static _getOneTranslationalVelocityAfterCollision(massA, posA, velocityA, massB, posB, velocityB) {
+        //TODO: refactor variables names to whatever mr boss boy tells me to
+        let partOne = (2 * massB) / (massA + massB);
+
+        let velocityDifference = [velocityA[0] - velocityB[0], velocityA[1] - velocityB[1]];
+        let positionDifference = [posA[0] - posB[0], posA[1], posB[1]];
+        let velDiffPosDiffDotProduct = MitoMathHelper._dotProduct(velocityDifference, positionDifference);
+        let positionMagnitudeSquared = MitoMathHelper._getMagnitudeSquared(positionDifference);
+
+        let partTwo = partOne * velDiffPosDiffDotProduct / positionMagnitudeSquared;
```

Comment:
I actually thing partOne and partTwo is fine since its math, but normally for logic it wouldn't be